oauth2.go

```diff
@@ -268,7 +268,12 @@ type ScopeString string
 
 // BuildScope is used to generate a scope string for oAuth2 flow
 func BuildScope(service Service, scope Scope, method Method, operation Operation) ScopeString {
-	built := fmt.Sprintf("%s.%s.%s", service, scope, method)
+	var built string = ""
+	if method != "" {
+		built += fmt.Sprintf("%s.%s.%s", service, scope, method)
+	} else {
+		built += fmt.Sprintf("%s.%s", service, scope)
+	}
 	if operation != "" {
 		built += "." + string(operation)
 	}
```

Comment:
Instead of creating an `else` line, See what we do with operation. This should be fine but could lead to weird errors if users are being silly.

```
built := fmt.Sprintf("%s.%s", service, scope)
if method != "" {
    built += fmt.Sprintf(".%s", method)
}
if operation != "" {
    built += fmt.Sprintf(".%s", operation)
}
```

Reply:
Sure, the suggested code snippet makes more sense.